components/scheduler/statemachine.go

```diff
@@ -0,0 +1,96 @@
+package scheduler
+
+import (
+	"fmt"
+	enumspb "go.temporal.io/api/enums/v1"
+	enumsspb "go.temporal.io/server/api/enums/v1"
+	schedspb "go.temporal.io/server/api/schedule/v1"
+	"go.temporal.io/server/common/persistence/serialization"
+	"go.temporal.io/server/service/history/hsm"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"time"
+)
+
+// Unique type identifier for this state machine.
+var StateMachineType = hsm.MachineType{
+	ID:   5,
```

Comment:
Yeah agreed --- it's very easy for future developers to accidentally duplicate an ID and afaik there is no guardrails against that it will just blow up somewhere else. 

Maybe use some sort of generated ID or type identities? Alternatively a string is probably better, but the duplicate problem technically remains.

Reply:
I think it's fine as long as all of the types are contained to a single codebase, it becomes an issue if we want to allow third party components.

My thinking for the future was to solve this by having a type database that assigns IDs. But that would mean that the type IDs shouldn't be in code anyways.